File: 03.CPP/1480.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> runningSum(std::vector<int>& nums) {
        int n = nums.size();

        std::vector<int> ans(n);
        ans[0] = nums[0];

        for (auto i = 1; i < n; i++) {
            ans[i] = ans[i - 1] + nums[i];
        }

        return ans;
    }
};

void printVector(std::vector<int>& arr) {
    std::cout << "[";
    for (int i = 0; i < arr.size(); i++) {
        std::cout << arr[i];

        if (i < arr.size() - 1 ) {
            std::cout << ", ";
        }
    }
    std::cout << "]";
}

int main() {
    Solution sol;

    std::vector<int> nums = {1, 2, 3, 4};
    std::vector<int> ans = sol.runningSum(nums);

    printVector(ans);
    return 0;
}

File: 01.Python/1480.py
from typing import List

class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        n = len(nums)

        ans = [0] * n
        ans[0] = nums[0]

        for i in range(1, n):
            ans[i] = ans[i - 1] + nums[i]

        return ans

sol = Solution()

nums = [1, 2, 3, 4]
result = sol.runningSum(nums)

print(result)